docs/deploy/Deploy-FhirConverterService.bicep

```diff
@@ -108,53 +126,72 @@ var securityAuthenticationAudiencesConfig = [for (audience, i) in securityAuthen
 }]
 
 var securityConfiguration = concat(securityEnabledConfiguration, securityEnabled ? concat(securityAuthenticationAuthorityConfig, securityAuthenticationAudiencesConfig) : [])
+var integrateTemplateStore = !empty(templateStorageAccountName) && !empty(templateStorageAccountContainerName)
 
 // Template hosting configuration
 var storageEnvironmentSuffix = az.environment().suffixes.storage
 var blobTemplateHostingConfigurationName = 'TemplateHosting__StorageAccountConfiguration__ContainerUrl'
 var blobTemplateHostingConfigurationValue = 'https://${templateStorageAccountName}.blob.${storageEnvironmentSuffix}/${templateStorageAccountContainerName}'
-var blobTemplateHostingConfiguration = [
+var blobTemplateHostingConfiguration = integrateTemplateStore ? [
   {
     name: blobTemplateHostingConfigurationName
     value: blobTemplateHostingConfigurationValue
   }
-]
+] : []
 
 // Application insights configuration
 var applicationInsightsConnectionStringConfigurationName = 'ConvertService__Telemetry__AzureMonitor__ApplicationInsightsConnectionString'
 var applicationInsightsUAMIClientIdConfigurationName = 'ConvertService__Telemetry__AzureMonitor__ManagedIdentityClientId'
-var telemetryConfiguration = [
+var telemetryConfiguration = configureApplicationInsights ? [
     {
         name: applicationInsightsConnectionStringConfigurationName
-        value: applicationInsightsConnectionString
+        secretRef: applicationInsightsConnectionStringSecretName
     }
     {
         name: applicationInsightsUAMIClientIdConfigurationName
-        value: applicationInsightsUAMIClientId
+        value: applicationInsightsUAMI.properties.clientId
     }
-]
+] : []
 
 // Environment Variables for Container App
-var envConfiguration =  concat(securityConfiguration, telemetryConfiguration, empty(templateStorageAccountName) ? [] : blobTemplateHostingConfiguration)
+var envConfiguration =  concat(securityConfiguration, telemetryConfiguration, blobTemplateHostingConfiguration)
 
 var imageName = 'healthcareapis/fhir-converter'
 
+// Configure identities
+var applicationInsightsUAMIResourceId = configureApplicationInsights ? applicationInsightsUAMI.id : ''
+var keyVaultUAMIResourceId = resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', keyVaultUAMIName)
+var userAssignedIdentities = configureApplicationInsights ? {
+  '${applicationInsightsUAMIResourceId}' : {}
+  '${keyVaultUAMIResourceId}' : {}

```

Comment:
Adding `keyVaultUAMIResourceId` should be decoupled from configureAppInsights and based off if the value was provided. Fine for now to leave, but in the future if we add another configuration which requires a secret, this will need to change.